Code/GraphMol/catch_adjustquery.cpp

```diff
@@ -0,0 +1,540 @@
+    CHECK(ps.makeAtomsGeneric == false);
+    CHECK(ps.makeBondsGeneric == false);
+    CHECK(ps.makeBondsGenericFlags == MolOps::ADJUST_IGNORENONE);
+
+    std::string json = R"JSON({"makeAtomsGeneric":true})JSON";
+    MolOps::parseAdjustQueryParametersFromJSON(ps, json);
+
+    CHECK(ps.makeAtomsGeneric == true);
+    CHECK(ps.makeBondsGeneric == false);
+    // the parsing updates the parameters, it doesn't replace them:
+
+    json = R"JSON({"makeBondsGeneric":true,
+      "makeBondsGenericFlags":"IGNOREDUMMIES|IGNORECHAINS"})JSON";
+    MolOps::parseAdjustQueryParametersFromJSON(ps, json);
+
+    CHECK(ps.makeAtomsGeneric == true);
+    CHECK(ps.makeBondsGeneric == true);
+    CHECK(ps.makeBondsGenericFlags ==
+          (MolOps::ADJUST_IGNOREDUMMIES | MolOps::ADJUST_IGNORECHAINS));
+  }
+  SECTION("useStereoCare") {
+    MolOps::AdjustQueryParameters ps;
+    CHECK(ps.useStereoCareForBonds == false);
+
+    std::string json = R"JSON({"useStereoCareForBonds":true})JSON";
+    MolOps::parseAdjustQueryParametersFromJSON(ps, json);
+    CHECK(ps.useStereoCareForBonds == true);
+    json = R"JSON({"useStereoCareForBonds":false})JSON";
+    MolOps::parseAdjustQueryParametersFromJSON(ps, json);
+    CHECK(ps.useStereoCareForBonds == false);
+  }
+  SECTION("bogus contents") {
+    MolOps::AdjustQueryParameters ps;
+    CHECK(ps.adjustDegree == true);
+    CHECK(ps.adjustDegreeFlags ==
+          (MolOps::ADJUST_IGNOREDUMMIES | MolOps::ADJUST_IGNORECHAINS));
+
+    std::string json = R"JSON({"bogosity":true})JSON";
+    MolOps::parseAdjustQueryParametersFromJSON(ps, json);
+    CHECK(ps.adjustDegree == true);
+
+    json = R"JSON({"adjustDegree":"foo"})JSON";
+    MolOps::parseAdjustQueryParametersFromJSON(ps, json);
+    CHECK(ps.adjustDegree == true);
+
+    json = R"JSON({"adjustDegreeFlags":"IGNORENONE|bogus"})JSON";
+    // clang-format off
+    CHECK_THROWS_AS(MolOps::parseAdjustQueryParametersFromJSON(ps, json),ValueErrorException);
+  }
+}
+
+
+TEST_CASE("MDL five-rings") {
+  MolOps::AdjustQueryParameters ps = MolOps::AdjustQueryParameters::noAdjustments();
+  ps.setMDLFiveRingAromaticity = true;
+  SECTION("query details") {
+    std::vector<std::tuple<std::string,std::string,std::string>> examples = {
+    {"adjustqueryprops_MDLfivering_1.mol","[#7H]1:[#6]:[#6]:[#6]:[#6]:1","[#7H]1:[#6]:[#6]:[#6]:[#6]:1"},
+    {"adjustqueryprops_MDLfivering_2.mol","[!#6&!#1]1:[#6]:[#6]:[#6]:[#6]:1","[!#6&!#1]1:[#6]:[#6]:[#6]:[#6]:1"},
+    {"adjustqueryprops_MDLfivering_3.mol","[!#1]1:[#6]:[#6]:[#6]:[#6]:1","[!#1]1-,:[#6]=,:[#6]-,:[#6]=,:[#6]-,:1"},
+    // NOTE that this is not technically correct according to the documentation, but if we make the bridging bond
+    // aromatic then it won't match azulene in a normal RDKit molecule, which is certainly not the intent of this.
+    {"adjustqueryprops_MDLfivering_4.mol","[#6]12:[#6]:[#6]:[#6]:[#6]-1:[#6]:[#6]:[#6]:[#6]:[#6]:2","[#6]12:[#6]:[#6]:[#6]:[#6]-1:[#6]:[#6]:[#6]:[#6]:[#6]:2"},
+    };
```

Comment:
These are a bit difficult to read, maybe it would be easier on the eye to do something like:

```suggestion
    std::string example1_smarts{"[#7H]1:[#6]:[#6]:[#6]:[#6]:1"};
    std::string example2_smarts{"[!#6&!#1]1:[#6]:[#6]:[#6]:[#6]:1"};
    std::string example3_smarts_no_adj{"[!#1]1:[#6]:[#6]:[#6]:[#6]:1"};
    std::string example3_smarts_5ring_aromatic{"[!#1]1-,:[#6]=,:[#6]-,:[#6]=,:[#6]-,:1"};
    std::string example4_smarts{"[#6]12:[#6]:[#6]:[#6]:[#6]-1:[#6]:[#6]:[#6]:[#6]:[#6]:2"};    
    
    std::vector<std::tuple<std::string,std::string,std::string>> examples = {
    {"adjustqueryprops_MDLfivering_1.mol",example1_smarts,example1_smarts},
    {"adjustqueryprops_MDLfivering_2.mol",example2_smarts,example2_smarts},
    {"adjustqueryprops_MDLfivering_3.mol",example3_smarts_no_adj,example3_smarts_5ring_aromatic},
    // NOTE that this is not technically correct according to the documentation, but if we make the bridging bond
    // aromatic then it won't match azulene in a normal RDKit molecule, which is certainly not the intent of this.
    {"adjustqueryprops_MDLfivering_4.mol",example4_smarts,example4_smarts}
    };
```

This way it is a bit easier to see for which examples we expect the same result, and which one changes.

Reply:
agreed that the formatting and the way they are written makes these difficult to read. Will try and alternate approach.